test.js

```diff
@@ -67,11 +99,96 @@ describe('minimist', function () {
     cli.use(set('eee', 'fff'))
 
     cli.on('end', function (argv) {
+      assert.equal(argv.aaa, 'bbb');
+      assert.equal(argv.ccc, 'ddd');
+      assert.equal(argv.eee, 'fff');
       i++;
     });
 
-    cli.parse(['a', 'b', 'c', '--foo=bar'], function (err, res) {
+    cli.parse(['a', 'b', 'c', '--foo=bar'], function (err, argv) {
+      assert.ifError(err);
       assert.equal(i, 1);
+      assert.equal(argv.aaa, 'bbb');
+      assert.equal(argv.ccc, 'ddd');
+      assert.equal(argv.eee, 'fff');
+      done();
+    });
+  });
+
+  it('should done callback be optional', function (done) {
+    var res = cli.parse(['a', 'b']);
+    assert.equal(typeof res.minimist, 'function');
+    done();
+  });
+
+  it('should result have parsed `.argv` property', function (done) {
+    var res = cli.parse(['--foo=bar', '--baz=qux']);
+    assert.equal(typeof res.minimist, 'function');
+    assert.equal(res.argv.foo, 'bar');
+    assert.equal(res.argv.baz, 'qux');
+    done();
+  });
+
+  it('should result have processed `.argv` property', function (done) {
+    cli.use(function () {
+      return function (argv, next) {
+        argv.foo = 'bar2';
+        argv.baz = 'qux2';
+        next(null, argv);
+      };
+    });
+    var res = cli.parse(['--foo=bar', '--baz=qux']);
```

Comment:
> do any of the tests without a callback have chained plugins?

hm. good point, i'll check.

> how will errors be handled if a plugin uses the callback (properly) to pass the error to the done function?

If there's no callback, it would seems that he don't want to handle them. It's logical, if he as man who build the cli app, write plugins in async way (with proper `next` use) he also will pass done callback and will expect errors and result there.

If he write sync plugins, that means he may want the syncness all way down and he can expect that it's not needed to pass done callback to end his job and receive same results as with callback.

All that changes just set sync aspect of the things.